### PR TITLE
Notify a Scrollable that the associated animation has stopped

### LINKED ISSUE #3675: Assertion failure when rotating screen during animation
1) Run Stocks and scroll the list
2) Tap on the Android task switcher while the list is animating
3) Rotate the screen
4) Resume the Stocks app

This triggers an assertion failure:

I/flutter (17668): The following assertion was thrown during performLayout():
I/flutter (17668): 'package:flutter/src/widgets/scrollable.dart': Failed assertion: line 389: '_controller.isAnimating
I/flutter (17668): || _simulation == null' is not true.
I/flutter (17668): Either the assertion indicates an error in the framework itself, or we should provide substantially
I/flutter (17668): more information in this error message to help you determine and fix the underlying cause.
I/flutter (17668): In either case, please report this assertion by filing a bug on GitHub:
I/flutter (17668):   https://github.com/flutter/flutter/issues/new
I/flutter (17668): The following RenderObject was being processed when the exception was fired:
I/flutter (17668):   RenderList NEEDS-LAYOUT
I/flutter (17668): This RenderObject had the following creator information:
I/flutter (17668):   ListViewport ← Listener ← _GestureSemantics ← RawGestureDetector-[GlobalKey 623875018] ←
I/flutter (17668):   ScrollableList-['stock-list'] ← StockList ← Container-['StockHomeTab.market'] ←
I/flutter (17668):   KeyedSubtree-['['StockHomeTab.market']'] ← RepaintBoundary-['['['StockHomeTab.market']']'] ←
I/flutter (17668):   PageViewport ← ⋯
I/flutter (17668): This RenderObject had the following descendants (showing up to depth 5):
I/flutter (17668):   RenderRepaintBoundary
I/flutter (17668):     RenderSemanticsGestureHandler
I/flutter (17668):       RenderPointerListener
I/flutter (17668):         RenderDecoratedBox
I/flutter (17668):           RenderPadding
I/flutter (17668):   RenderRepaintBoundary NEEDS-LAYOUT
I/flutter (17668):     RenderSemanticsGestureHandler NEEDS-LAYOUT
I/flutter (17668):       RenderPointerListener NEEDS-LAYOUT
I/flutter (17668):         RenderDecoratedBox NEEDS-LAYOUT
I/flutter (17668):           RenderPadding NEEDS-LAYOUT
I/flutter (17668):   RenderRepaintBoundary NEEDS-LAYOUT
I/flutter (17668):     RenderSemanticsGestureHandler NEEDS-LAYOUT
I/flutter (17668):       RenderPointerListener NEEDS-LAYOUT
I/flutter (17668):         RenderDecoratedBox NEEDS-LAYOUT
I/flutter (17668):           RenderPadding NEEDS-LAYOUT
I/flutter (17668):   RenderRepaintBoundary NEEDS-LAYOUT
I/flutter (17668):     RenderSemanticsGestureHandler NEEDS-LAYOUT
I/flutter (17668):       RenderPointerListener NEEDS-LAYOUT
I/flutter (17668):         RenderDecoratedBox NEEDS-LAYOUT
I/flutter (17668):           RenderPadding NEEDS-LAYOUT
I/flutter (17668): When the exception was thrown, this was the stack:
I/flutter (17668): #0      _AssertionError._throwNew (dart:core-patch/errors_patch.dart:27)
I/flutter (17668): #1      ScrollableState.didUpdateScrollBehavior (package:flutter/src/widgets/scrollable.dart:389)
I/flutter (17668): #2      _ScrollableListState._handleExtentsChanged.<anonymous closure> (package:flutter/src/widgets/scrollable_list.dart:96)
I/flutter (17668): #3      State.setState (package:flutter/src/widgets/framework.dart:419)
I/flutter (17668): #4      _ScrollableListState._handleExtentsChanged (package:flutter/src/widgets/scrollable_list.dart:95)
I/flutter (17668): #5      _VirtualListViewportElement.layout (package:flutter/src/widgets/scrollable_list.dart:285)
I/flutter (17668): #6      RenderObject.invokeLayoutCallback (package:flutter/src/rendering/object.dart:1348)
I/flutter (17668): #7      RenderList.performLayout (package:flutter/src/rendering/list.dart:138)
I/flutter (17668): #8      RenderObject.layout (package:flutter/src/rendering/object.dart:1268)
I/flutter (17668): #9      RenderProxyBox.performLayout (package:flutter/src/rendering/proxy_box.dart:80)
I/flutter (17668): #10     RenderObject.layout (package:flutter/src/rendering/object.dart:1268)
I/flutter (17668): #11     RenderProxyBox.performLayout (package:flutter/src/rendering/proxy_box.dart:80)
...

### PR DESCRIPTION
**Pull request by jason-simmons**
This also required changing the AnimationController state transition
logic to signal completion of the animation during the tick that
finishes the simulation.

Fixes https://github.com/flutter/flutter/issues/3675

### CODE REVIEW

**Comment by jason-simmons** [comment]
@abarth @Hixie

File: examples/flutter_gallery/test/scrollable_dispose_test.dart [hunk 1]
```diff
@@ -0,0 +1 @@
+../../../packages/flutter/test/widget/scrollable_dispose_test.dart
```
Comment (Hixie) [inline examples/flutter_gallery/test/scrollable_dispose_test.dart]: this shouldn't be included :-)

File: packages/flutter/test/widget/scrollable_dispose_test.dart [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter_test/flutter_test.dart';
+import 'package:flutter/widgets.dart';
+
+import 'test_widgets.dart';
+
+void main() {
+  testWidgets('fling, fling generates two start/end pairs', (WidgetTester tester) async {
```
Comment (Hixie) [inline packages/flutter/test/widget/scrollable_dispose_test.dart]: update the test name

**Comment by Hixie** [comment]
@abarth you should review this since I helped write it

File: examples/flutter_gallery/test/scrollable_dispose_test.dart (hunk 1 above)
Comment (jason-simmons) [inline examples/flutter_gallery/test/scrollable_dispose_test.dart]: done

File: packages/flutter/test/widget/scrollable_dispose_test.dart (hunk 2 above)
Comment (jason-simmons) [inline packages/flutter/test/widget/scrollable_dispose_test.dart]: done

**Comment by abarth** [comment]
LGTM

**Pull request merged**